### ISSUE TITLE
dosbox-staging 0.80.1 stopped building with MacPorts on newest MacOSX 14 Sonoma (Xcode 15).

### ISSUE THREAD

**Issue by thaewrapt** [dosbox-staging/dosbox-staging#2967]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### What Operating System are you using?

macOS (Apple Silicon)

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

During build, this particular error leads to a failure: 

```
:info:build /usr/bin/clang -Isrc/libs/manymouse/libmanymouse.a.p -Isrc/libs/manymouse -I../dosbox-staging-0.80.1/src/libs/manymouse -I../dosbox-staging-0.80.1/include -I. -I../dosbox-staging-0.80.1 -I/opt/local/include -fcolor-diagnostics
 -DNDEBUG -Wall -Winvalid-pch -Wextra -Wpedantic -O3 -Wno-unknown-pragmas -Weffc++ -Wextra-semi -fstrict-aliasing -Wstrict-aliasing -fmerge-all-constants -fno-math-errno -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal
-math -ffinite-math-only -ffunction-sections -fdata-sections -fPIC -pipe -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -arch arm64 -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -MD -MQ src/libs/manymo
use/libmanymouse.a.p/x11_xinput2.c.o -MF src/libs/manymouse/libmanymouse.a.p/x11_xinput2.c.o.d -o src/libs/manymouse/libmanymouse.a.p/x11_xinput2.c.o -c ../dosbox-staging-0.80.1/src/libs/manymouse/x11_xinput2.c
:info:build ../dosbox-staging-0.80.1/src/libs/manymouse/x11_xinput2.c:377:13: error: call to undeclared function 'select'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
:info:build         if (select(fd+1, &fdset, NULL, NULL, &nowait) == 1)
:info:build             ^
:info:build 1 error generated.
```

All the additional information and a full build log is in the MacPorts ticket: https://trac.macports.org/ticket/68277

### Steps to reproduce the behaviour.

Explain how to reproduce
1.  Run `sudo port install dosbox-staging` on MacOSX 14 Sonoma.
2. Confirm installing additional dependencies if applicable.
3. Build fails.
4.  Check the log mentioned in the error message of the command.

### Provide a Log

```text
https://trac.macports.org/attachment/ticket/68277/dosbox-staging-main.log
```

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by weirddan455** [comment]
I'm not a Mac user so I don't know anything about MacPorts.  However, that file is not supposed to be built by Mac.  It's for X11 so usually you'd only want it for Linux and BSD and such.

https://github.com/dosbox-staging/dosbox-staging/blob/8b5b724839b23f61e7474d331c1bdd76087665e0/src/libs/manymouse/x11_xinput2.c#L20-L31

Maybe the Apple part of this macro needs to be re-written.  As a workaround, you can probably just change the `#if SUPPORT_XINPUT2` at line 31 to `#if 0` to compile it out.

Alternatively, the actual error you're seeing is just a missing header file.  You could try adding `#include <sys/select.h>` and see if that lets it compile.

**Comment by kcgen** [comment]
Some environments offer X-like user-space stacks, like MSYS2, Cygwin, and (maybe) MacPorts (if they allow compiling lots of X11 programs).

In `meson.build` to test for the xinput 2.0 library availability, regardless of the OS:

```python
# ManyMouse optionally supports the X Input 2.0 protocol (regardless of OS)
xinput2_dep = optional_dep
xinput2_required = (get_option('use_xinput2') == 'true')
if xinput2_required or (get_option('use_xinput2') == 'auto')
    xinput2_dep = dependency(
        'xi',
        version: ['>= 1.4', '< 2'],
        required: xinput2_required,
        include_type: 'system',
    )
endif
conf_data.set10('SUPPORT_XINPUT2', xinput2_dep.found())
```

@thaewrapt, when you run `meson setup build`, do you see:

```
Run-time dependency xi found: YES 1.8
```

(it's near the end of the setup phase)

If yes: then you've got the library, so we should try to let the software build with it by fixing headers.

**Comment by thaewrapt** [comment]
> @thaewrapt, when you run `meson setup build`, do you see:
> 
> ```
> Run-time dependency xi found: YES 1.8
> ```

Yeah, it's in the logfile (the one I've mentioned in the issue):
```
:info:configure Run-time dependency xi found: YES 1.8.1
```

**Comment by thaewrapt** [comment]
Would it be more comfortable if I try a build w/o MacPorts and attach a new logfile instead? So we wouldn't have a noise (which is certainly useful for MacPorts maintainers but less so to you).

**Comment by kcgen** [comment]
> > @thaewrapt, when you run `meson setup build`, do you see:
> > ```
> > Run-time dependency xi found: YES 1.8
> > ```
> 
> Yeah, it's in the logfile (the one I've mentioned in the issue):
> 
> ```
> :info:configure Run-time dependency xi found: YES 1.8.1
> ```

That's great MacPorts has xinput!

This means `SUPPORT_XINPUT2` will be defined when compiling many mouse.

I think @weirddan455's suggestion is the best approach:

>  the actual error you're seeing is just a missing header file. You could try adding #include <sys/select.h> and see if that lets it compile.

On line 38 in `src/libs/manymouse/x11_xinput2.c`, insert `#include <sys/select.h>`, like follows:

```c++
#if SUPPORT_XINPUT2

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <dlfcn.h>
#include <X11/extensions/XInput2.h>
#include <sys/select.h>
```

**Comment by thaewrapt** [comment]
Can confirm that is builds manually with a proposed change.

**Comment by thaewrapt** [comment]
PR sent.

**Comment by weirddan455** [comment]
Fixed in #2968
